packages/token/src/tests/erc2981/test_erc2981_accesscontrol.cairo

```diff
@@ -0,0 +1,419 @@
+use crate::common::erc2981::ERC2981Component::ROYALTY_ADMIN_ROLE;
+use crate::common::erc2981::ERC2981Component;
+use crate::common::erc2981::interface::{IERC2981ABIDispatcher, IERC2981ABIDispatcherTrait};
+use openzeppelin_access::accesscontrol::interface::{
+    AccessControlABIDispatcher, AccessControlABIDispatcherTrait
+};
+use openzeppelin_testing as utils;
+use openzeppelin_testing::constants::{OWNER, OTHER_ADMIN, OTHER_ROLE, OTHER, ZERO, RECIPIENT};
+use openzeppelin_utils::serde::SerializedAppend;
+use snforge_std::start_cheat_caller_address;
+use starknet::{ContractAddress, contract_address_const};
+
+fn DEFAULT_RECEIVER() -> ContractAddress {
+    contract_address_const::<'DEFAULT_RECEIVER'>()
+}
+
+// 0.5% (default denominator is 10000)
+const DEFAULT_FEE_NUMERATOR: u128 = 50;
+// 5% (default denominator is 10000)
+const FEE_NUMERATOR: u128 = 500;
+
+fn setup_dispatcher() -> IERC2981ABIDispatcher {
```

Comment:
Indeed, my initial intention was to use COMPONENT_STATE, but then I understood that it's better to go with dispatcher because:
- the newly introduced traits are specifically designed to provide external functions
- external functions mostly have the same names as their internal counterparts and this approach makes it more apparent that the external functions are used

Reply:
Building mocks makes a difference here indeed. I refactored the tests to use COMPONENT_STATE and underscored the internal functions to prevent names from colliding